mmv1/third_party/tgc/tests/data/example_iam_deny_policy.json

```diff
@@ -0,0 +1,41 @@
+[
+  {
+    "name": "//iam.googleapis.com/policies/cloudresourcemanager.googleapis.com%2Fprojects%2F{{.Provider.project}}/denypolicies/my-deny-policy",
+    "asset_type": "iam.googleapis.com/DenyPolicy",
```

Comment:
The asset type should be [iam.googleapis.com/PolicyV2](https://cloud.google.com/iam/docs/reference/rest/v2/policies). 

Maybe the generated asset type in `iam2.ResourceConverterIAM2DenyPolicy()` is wrong.

Reply:
You're right, the CAI name is PolicyV2, it's just more commonly referred to as a DenyPolicy because V2 IAM policies can only be configured as deny policies. I also realized the converter is not setting the `name` field like it should.

What is the path forward? Does this require a handwritten converter?

Reply:
What should the `name` field be?

Reply:
Looking at a real CAI export, `name` should be `policies/cloudresourcemanager.googleapis.com%2Fprojects%2F{{project_number}}/denypolicies/{{tf_name_field}}` but in a TF plan the project number might not be known so using project ID instead is fine. There is also a `kind` which should be `DenyPolicy` but I don't care about that field for my use case since all V2 policies are of DenyPolicy kind.

Reply:
When generating the cai asset type, the resource name is used.
https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/templates/tgc/resource_converter.go.tmpl#L50


We can add another field `CaiName` or `TgcName` to https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/api/resource.go#L39. 


If CaiName is provided, it is used to generate the cai asset type. Otherwise, the resource name is used.